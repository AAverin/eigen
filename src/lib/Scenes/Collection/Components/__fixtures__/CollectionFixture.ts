export const CollectionFixture = {
  slug: "street-art-now",
  id: "top-level-id",
  title: "Street Art Now",
  descriptionMarkdown:
    "A beach towel by [Yayoi Kusama](https://www.artsy.net/artist/yayoi-kusama), a classic print by [Alexander Calder](https://www.artsy.net/artist/alexander-calder), or a piggy bank by [Yoshitomo Nara](https://www.artsy.net/artist/yoshitomo-nara)â€”there are many ways to give the gift of art. Below, discover creative presents that your loved ones will treasure for years to come.",
  headerImage: "http://imageuploadedbymarketingteam.jpg",
  image: {
    id: "another-id",
    edges: [
      {
        node: { id: "an-id", image: { resized: { url: "https://defaultmostmarketableartworkincollectionimage.jpg" } } },
      },
    ],
  },
  collectionArtworks: {
    id: "QXJ0d29yazpjYXJvbHluLWJyYWR5LXN0b3VyaGVhZC1hY3Jvc3MtdGhlLWxha2U",
    pageInfo: {
      startCursor: "YXJyYXljb25uZWN0aW9uOjA=",
      endCursor: "YXJyYXljb25uZWN0aW9uOjc=",
      hasNextPage: false,
    },
    edges: [
      {
        cursor: "fds",
        id: "fdas",
        node: {
          __typename: "Artwork",
          id: "QXJ0d29yazpjYXJvbHluLWJyYWR5LXN0b3VyaGVhZC1hY3Jvc3MtdGhlLWxha2U=",
          slug: "carolyn-brady-stairheads-across-the-lake",
          image: {
            aspectRatio: 1.53,
            aspect_ratio: 1.53,
            url: "https://d32dm0rphc51dk.cloudfront.net/tH19ZfyCDocnNsKzXOZfHg/medium.jpg",
          },
          title: "Stourhead Across the Lake",
          date: "1997",
          sale_message: "$15,000",
          is_biddable: false,
          is_acquireable: false,
          is_offerable: false,
          sale: null,
          sale_artwork: null,
          artists: [
            {
              id: "carolyn-brady",
              name: "Carolyn Brady",
            },
          ],
          partner: {
            name: "Anderson Fine Art Gallery",
            id: "anderson-fine-art-gallery",
          },
          href: "/artwork/carolyn-brady-scoured-across-the-lake",
        },
      },
      {
        cursor: "fds",
        id: "fdas",
        node: {
          __typename: "Artwork",
          id: "QXJ0d29yazpjYXJvbHluLWJyYWR5LXN0b3VyaGVhZC1hY3Jvc3MtdGhlLWxha2U=",
          slug: "carolyn-brady-stairheads-across-the-lake",
          image: {
            aspectRatio: 1.53,
            aspect_ratio: 1.53,
            url: "https://d32dm0rphc51dk.cloudfront.net/tH19ZfyCDocnNsKzXOZfHg/medium.jpg",
          },
          title: "Stourhead Across the Lake",
          date: "1997",
          sale_message: "$15,000",
          is_biddable: false,
          is_acquireable: false,
          is_offerable: false,
          sale: null,
          sale_artwork: null,
          artists: [
            {
              id: "carolyn-brady",
              name: "Carolyn Brady",
            },
          ],
          partner: {
            name: "Anderson Fine Art Gallery",
            id: "anderson-fine-art-gallery",
          },
          href: "/artwork/carolyn-brady-scoured-across-the-lake",
        },
      },
      {
        cursor: "fds",
        id: "fdas",
        node: {
          __typename: "Artwork",
          id: "QXJ0d29yazpjYXJvbHluLWJyYWR5LXN0b3VyaGVhZC1hY3Jvc3MtdGhlLWxha2U=",
          slug: "carolyn-brady-stairheads-across-the-lake",
          image: {
            aspectRatio: 1.53,
            aspect_ratio: 1.53,
            url: "https://d32dm0rphc51dk.cloudfront.net/tH19ZfyCDocnNsKzXOZfHg/medium.jpg",
          },
          title: "Stourhead Across the Lake",
          date: "1997",
          sale_message: "$15,000",
          is_biddable: false,
          is_acquireable: false,
          is_offerable: false,
          sale: null,
          sale_artwork: null,
          artists: [
            {
              id: "carolyn-brady",
              name: "Carolyn Brady",
            },
          ],
          partner: {
            name: "Anderson Fine Art Gallery",
            id: "anderson-fine-art-gallery",
          },
          href: "/artwork/carolyn-brady-scoured-across-the-lake",
        },
      },
      {
        cursor: "fds",
        id: "fdas",
        node: {
          __typename: "Artwork",
          id: "QXJ0d29yazpjYXJvbHluLWJyYWR5LXN0b3VyaGVhZC1hY3Jvc3MtdGhlLWxha2U=",
          slug: "carolyn-brady-stairheads-across-the-lake",
          image: {
            aspectRatio: 1.53,
            aspect_ratio: 1.53,
            url: "https://d32dm0rphc51dk.cloudfront.net/tH19ZfyCDocnNsKzXOZfHg/medium.jpg",
          },
          title: "Stourhead Across the Lake",
          date: "1997",
          sale_message: "$15,000",
          is_biddable: false,
          is_acquireable: false,
          is_offerable: false,
          sale: null,
          sale_artwork: null,
          artists: [
            {
              id: "carolyn-brady",
              name: "Carolyn Brady",
            },
          ],
          partner: {
            name: "Anderson Fine Art Gallery",
            id: "anderson-fine-art-gallery",
          },
          href: "/artwork/carolyn-brady-scoured-across-the-lake",
        },
      },
      {
        cursor: "fds",
        id: "fdas",
        node: {
          __typename: "Artwork",
          id: "QXJ0d29yazpjYXJvbHluLWJyYWR5LXN0b3VyaGVhZC1hY3Jvc3MtdGhlLWxha2U=",
          slug: "carolyn-brady-stairheads-across-the-lake",
          image: {
            aspectRatio: 1.53,
            aspect_ratio: 1.53,
            url: "https://d32dm0rphc51dk.cloudfront.net/tH19ZfyCDocnNsKzXOZfHg/medium.jpg",
          },
          title: "Stourhead Across the Lake",
          date: "1997",
          sale_message: "$15,000",
          is_biddable: false,
          is_acquireable: false,
          is_offerable: false,
          sale: null,
          sale_artwork: null,
          artists: [
            {
              id: "carolyn-brady",
              name: "Carolyn Brady",
            },
          ],
          partner: {
            name: "Anderson Fine Art Gallery",
            id: "anderson-fine-art-gallery",
          },
          href: "/artwork/carolyn-brady-scoured-across-the-lake",
        },
      },
      {
        cursor: "fds",
        id: "fdas",
        node: {
          __typename: "Artwork",
          id: "QXJ0d29yazpjYXJvbHluLWJyYWR5LXN0b3VyaGVhZC1hY3Jvc3MtdGhlLWxha2U=",
          slug: "carolyn-brady-stairheads-across-the-lake",
          image: {
            aspectRatio: 1.53,
            aspect_ratio: 1.53,
            url: "https://d32dm0rphc51dk.cloudfront.net/tH19ZfyCDocnNsKzXOZfHg/medium.jpg",
          },
          title: "Stourhead Across the Lake",
          date: "1997",
          sale_message: "$15,000",
          is_biddable: false,
          is_acquireable: false,
          is_offerable: false,
          sale: null,
          sale_artwork: null,
          artists: [
            {
              id: "carolyn-brady",
              name: "Carolyn Brady",
            },
          ],
          partner: {
            name: "Anderson Fine Art Gallery",
            id: "anderson-fine-art-gallery",
          },
          href: "/artwork/carolyn-brady-scoured-across-the-lake",
        },
      },
    ],
  },
}
