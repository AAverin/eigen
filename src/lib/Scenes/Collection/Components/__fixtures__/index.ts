export const CollectionFixture = {
  collection: {
    title: "Street Art Now",
    description:
      '<p>From <a href="https://www.someartsyimage.net/artist/jean-michel-basquiat">Jean-Michel Basquiat</a> to <a href="https://www.someartsyimage.net/artist/shepard-fairey">Shepard Fairey</a>, street artists are known for their creative expressions of rebellion.</p>',
    headerImage: "http://imageuploadedbymarketingteam.jpg",
    image: { edges: [{ node: { imageUrl: "https://defaultmostmarketableartworkincollectionimage.jpg" } }] },
  },
}
