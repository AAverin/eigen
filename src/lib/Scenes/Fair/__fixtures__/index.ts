export const FairFixture = {
  id: "sofa-chicago-2018",
  name: "SOFA CHICAGO 2018",
  image: {
    image_url: "https://d32dm0rphc51dk.cloudfront.net/o0Zrtm-CgwbV1syD_33DuA/:version.jpg",
    aspect_ratio: 1,
    url: "https://d32dm0rphc51dk.cloudfront.net/o0Zrtm-CgwbV1syD_33DuA/square.jpg",
  },
  organizer: {
    profile: {
      icon: {
        id: "58ff95298b3b8104783c119c",
        href: null,
        height: 279,
        width: 279,
        url: "https://d32dm0rphc51dk.cloudfront.net/B1x-CqoWO1KitQnRj7PrrA/untouched-png.png",
      },
      name: "SOFA CHICAGO",
      __id: "UHJvZmlsZTpzb2ZhLWNoaWNhZ28=",
    },
  },
  start_at: "2018-11-01T17:00:00+00:00",
  end_at: "2018-11-04T18:00:00+00:00",
  location: {
    address: "",
    address_2: "",
    coordinates: {
      lat: 41.891551,
      lng: -87.6073751,
    },
    display: "41.891972, -87.605161",
    city: "Chicago",
    state: "Illinois",
    country: "United States",
    postal_code: "60611",
  },
  __id: "RmFpcjpzb2ZhLWNoaWNhZ28tMjAxOA==",
}
