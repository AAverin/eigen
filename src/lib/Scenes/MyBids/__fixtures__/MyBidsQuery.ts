export const sales = {
  edges: [
    {
      node: {
        saleType: "auction",
        href: "/auction/swann-auction-galleries-lgbtq-plus-art-material-culture-and-history",
        endAt: null,
        liveStartAt: "2020-08-13T16:00:00+00:00",
        displayTimelyAt: "live in 10d",
        timeZone: "America/New_York",
        name: "Swann Auction Galleries: LGBTQ+ Art, Material Culture & History",
        slug: "swann-auction-galleries-lgbtq-plus-art-material-culture-and-history",
        coverImage: {
          url: "https://d32dm0rphc51dk.cloudfront.net/1tJKlnBgHGvZSCZPlAvwAQ/wide.jpg",
        },
        partner: {
          name: "Swann Auction Galleries",
        },
      },
    },
    {
      node: {
        saleType: "auction",
        href: "/auction/heritage-urban-art-summer-skate",
        endAt: null,
        liveStartAt: "2020-08-05T15:00:00+00:00",
        displayTimelyAt: "live in 2d",
        timeZone: "America/Chicago",
        name: "Heritage: Urban Art Summer Skate",
        slug: "heritage-urban-art-summer-skate",
        coverImage: {
          url: "https://d32dm0rphc51dk.cloudfront.net/JOeiPjbfKixGJbQjQHubXA/source.jpg",
        },
        partner: {
          name: "Heritage Auctions",
        },
      },
    },
  ],
}

export const lotStandings = [
  {
    // Reserve not met
    isHighestBidder: false,
    isLeadingBidder: true,
    sale: {
      displayTimelyAt: "in progress",
      liveStartAt: "2020-08-03T19:10:10+00:00",
      endAt: null,
    },
    saleArtwork: {
      id: "U2FsZUFydHdvcms6NWYyODM2MzRmOTc5N2UwMDA3ZGM1MDU1",
      lotLabel: "3",
      counts: {
        bidderPositions: 1,
      },
      currentBid: {
        display: "CHF 1,800",
      },
      artwork: {
        artistNames: "Maskull Lasserre",
        href: "/artwork/maskull-lasserre-painting",
        image: {
          url: "https://d2v80f5yrouhh2.cloudfront.net/zrtyPc3hnFNl-1yv80qS2w/medium.jpg",
        },
      },
    },
  },
  {
    // Winning
    isHighestBidder: true,
    isLeadingBidder: true,
    sale: {
      displayTimelyAt: "in progress",
      liveStartAt: "2020-08-03T19:10:10+00:00",
      endAt: null,
    },
    saleArtwork: {
      id: "U2FsZUFydHdvcms6NWYyODM2MzNmOTc5N2UwMDA3ZGM1MDRk",
      lotLabel: "2",
      counts: {
        bidderPositions: 2,
      },
      currentBid: {
        display: "CHF 36,000",
      },
      artwork: {
        artistNames: "Zach Eugene Salinger-Simonson",
        href: "/artwork/zach-eugene-salinger-simonson-ennui",
        image: {
          url: "https://d2v80f5yrouhh2.cloudfront.net/NtVjXx1dzUIsOfFdyW0XZw/medium.jpg",
        },
      },
    },
  },
  {
    // Outbid
    isHighestBidder: false,
    isLeadingBidder: false,
    sale: {
      displayTimelyAt: "in progress",
      liveStartAt: "2020-08-03T19:10:10+00:00",
      endAt: null,
    },
    saleArtwork: {
      id: "U2FsZUFydHdvcms6NWYyODM2MzFmOTc5N2UwMDA3ZGM1MDQ1",
      lotLabel: "1",
      counts: {
        bidderPositions: 2,
      },
      currentBid: {
        display: "CHF 4,750",
      },
      artwork: {
        artistNames: "Leif Erik Nygards",
        href: "/artwork/leif-erik-nygards-surrealism",
        image: {
          url: "https://d2v80f5yrouhh2.cloudfront.net/dwqgENIWYbWFU_wntBtxFg/medium.jpg",
        },
      },
    },
  },
]
