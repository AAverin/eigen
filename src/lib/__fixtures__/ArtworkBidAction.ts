export const ArtworkFromAuctionPreview = {
  internalID: "artwork_from_preview_auction",
  sale: {
    registrationStatus: null,
    isPreview: true,
    isOpen: false,
    isLiveOpen: false,
    isClosed: false,
    isRegistrationClosed: false,
  },
  saleArtwork: {
    increments: [
      { cents: 400000, display: "CHF4,000" },
      { cents: 425000, display: "CHF4,250" },
      { cents: 450000, display: "CHF4,500" },
      { cents: 475000, display: "CHF4,750" },
    ],
  },
  myLotStanding: null,
}

export const ArtworkFromTimedAuctionRegistrationOpen = {
  internalID: "artwork_from_open_non_live_auction",
  sale: {
    registrationStatus: null,
    isPreview: false,
    isOpen: true,
    isLiveOpen: false,
    isClosed: false,
    isRegistrationClosed: false,
  },
  saleArtwork: {
    increments: [
      { cents: 90000, display: "$900" },
      { cents: 95000, display: "$950" },
      { cents: 100000, display: "$1,000" },
      { cents: 110000, display: "$1,100" },
    ],
  },
  myLotStanding: null,
}

export const ArtworkFromTimedAuctionRegistrationClosed = {
  internalID: "artwork_from_open_non_live_auction",
  sale: {
    registrationStatus: null,
    isPreview: false,
    isOpen: true,
    isLiveOpen: false,
    isClosed: false,
    isRegistrationClosed: true,
  },
  saleArtwork: {
    increments: [
      { cents: 750000, display: "$7,500" },
      { cents: 800000, display: "$8,000" },
      { cents: 850000, display: "$8,500" },
      { cents: 900000, display: "$9,000" },
    ],
  },
  myLotStanding: null,
}

export const ArtworkFromLiveAuctionRegistrationOpen = {
  internalID: "artwork_from_open_live_auction_open_registration",
  sale: {
    registrationStatus: null,
    isPreview: false,
    isOpen: true,
    isLiveOpen: true,
    isClosed: false,
    isRegistrationClosed: false,
  },
  saleArtwork: {
    increments: [
      { cents: 320000, display: "€3,200" },
      { cents: 350000, display: "€3,500" },
      { cents: 380000, display: "€3,800" },
      { cents: 400000, display: "€4,000" },
    ],
  },
  myLotStanding: null,
}

export const ArtworkFromLiveAuctionRegistrationClosed = {
  internalID: "artwork_from_open_live_auction_closed_registration",
  sale: {
    registrationStatus: null,
    isPreview: false,
    isOpen: true,
    isLiveOpen: true,
    isClosed: false,
    isRegistrationClosed: true,
  },
  saleArtwork: {
    increments: [
      { cents: 320000, display: "€3,200" },
      { cents: 350000, display: "€3,500" },
      { cents: 380000, display: "€3,800" },
      { cents: 400000, display: "€4,000" },
    ],
  },
  myLotStanding: null,
}

export const ArtworkFromClosedAuction = {
  internalID: "artwork_from_closed_auction",
  sale: {
    registrationStatus: null,
    isPreview: false,
    isOpen: false,
    isLiveOpen: false,
    isClosed: true,
    isRegistrationClosed: false,
  },
  saleArtwork: {
    increments: [
      { cents: 425000, display: "£4,250" },
      { cents: 450000, display: "£4,500" },
      { cents: 475000, display: "£4,750" },
      { cents: 500000, display: "£5,000" },
    ],
  },
  myLotStanding: null,
}

export const NotRegisteredToBid = {
  myLotStanding: null,
  sale: {
    registrationStatus: null,
  },
}

export const BidderPendingApproval = {
  myLotStanding: null,
  sale: {
    registrationStatus: {
      id: "bidder_pending_approval",
      qualifiedForBidding: false,
    },
  },
}

export const RegisteredBidder = {
  myLotStanding: null,
  sale: {
    registrationStatus: { id: "bidder_approved", qualifiedForBidding: true },
  },
}

export const RegistedBidderWithBids = {
  myLotStanding: [
    {
      mostRecentBid: {
        maxBid: {
          cents: 30000,
        },
      },
    },
  ],
  sale: {
    registrationStatus: { id: "bidder_approved", qualifiedForBidding: true },
  },
}
