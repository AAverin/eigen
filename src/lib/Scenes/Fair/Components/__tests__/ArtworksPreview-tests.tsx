import { renderRelayTree } from "lib/tests/renderRelayTree"
import { graphql } from "react-relay"
import { fairFixture } from "../../__fixtures__"
import { ArtworksPreviewContainer as ArtworksPreview } from "../ArtworksPreview"

jest.unmock("react-relay")

it("renders properly", async () => {
  const tree = await renderRelayTree({
    Component: ArtworksPreview,
    query: graphql`
      query ArtworksPreviewTestsQuery @raw_response_type {
        fair(id: "sofa-chicago-2018") {
          ...ArtworksPreview_fair
        }
      }
    `,
    mockData: {
      fair: fairFixture,
    },
  })

  expect(tree.html()).toMatchSnapshot()
})
