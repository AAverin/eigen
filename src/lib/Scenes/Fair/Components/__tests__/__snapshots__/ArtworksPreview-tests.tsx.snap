// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `"<text font-family=\\"AGaramondPro-Regular\\" font-size=\\"26px\\" lineheight=\\"32px\\" mt=\\"2\\" mb=\\"3\\" class=\\"Typography-sc-1pie2ao-0 ehDCiu\\">All works</text>"`;
